app/components/Search/Search.css

```diff
@@ -53,143 +53,109 @@
+  padding: 2rem 2rem 4rem;
+  background-color: var(--lego-card-color);
   border-radius: 0 0 var(--border-radius-lg) var(--border-radius-lg);
-
-  @media (--small-viewport) {
-    flex-direction: column;
-  }
 }
 
-.scrollAble {
+.sidePanel {
   display: flex;
-
-  @media (--small-viewport) {
-    flex-direction: column;
-  }
+  flex-direction: column;
+  gap: 1.5rem;
+  flex-grow: 1;
 }
 
-.navigationHeader {
+.quickLinksHeader {
   color: var(--lego-font-color);
-}
-
-.navigationFlex {
-  display: flex;
-  flex-wrap: wrap;
-}
-
-.navigationLink {
-  flex-basis: 33%;
-  min-width: 225px;
-  max-width: 100%;
-
-  @media (--mobile-device) {
-    flex-basis: 50%;
-  }
-
-  @media (--small-viewport) {
-    flex: 1;
-  }
+  padding-left: 15px;
 }
 
 .quickLinks {
-  flex: 1;
-  background-color: var(--lego-card-color);
-  color: var(--color-white);
-  padding: 2rem 2rem 3rem;
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(240px, 1fr));
+  list-style-type: none;
 }
 
-.quickLinks::after {
-  content: '';
-  display: block;
-  height: 2rem;
-  width: 100%;
+.quickLinkWrapper {
+  padding: 0;
 }
 
-.quickLinks li {
-  list-style: none;
-}
-
-.quickLinks a {
-  display: block;
+.quickLink {
   padding: 4px 15px;
+  box-sizing: border-box;
   font-size: var(--font-size-lg);
-  transition: background-color var(--linear-fast);
   border-radius: var(--border-radius-md);
+  transition: background-color var(--linear-fast);
 }
 
-.quickLinks a:hover {
+.quickLink:hover {
   background-color: var(--additive-background);
 }
 
-.results {
-  flex: 2;
-  padding: 2rem;
-  background: var(--lego-color-gray-light);
-}
-
-html[data-theme='dark'] .results {
-  background: var(--lego-card-color);
+.quickLinkContent {
+  gap: 0.4rem;
 }
 
-.resultDate {
-  color: var(--lego-font-color);
-  margin-left: auto;
-  white-space: nowrap;
-  padding-left: 5px;
-
-  @media (--small-viewport) {
-    display: none;
-  }
-}
-
-.resultDateMobile {
-  display: none;
-  margin-left: auto;
-
-  @media (--small-viewport) {
-    display: block;
-  }
-}
-
-.resultDetails {
-  color: var(--lego-font-color);
-  font-size: var(--font-size-sm);
-  display: flex;
-  justify-content: space-between;
-  line-height: 15px;
+.results {
+  flex: 1 1 66%;
+  min-width: 0;
 }
 
 .resultType {
+  display: block;
   text-transform: capitalize;
 }
 
 .resultItem {
-  border-bottom: 1px solid var(--border-gray);
-  padding: 0.5rem;
-  height: 64px;
   display: flex;
   align-items: center;
+  border-bottom: 1px solid var(--border-gray);
+  padding: 0.5rem;
+  height: 4rem;
 }
 
 .resultTitle {
-  display: table;
-  table-layout: fixed;
-  width: 100%;
-  line-height: 20px;
-  padding-top: 2px;
+  display: flex;
+  min-width: 0;
+  flex-direction: column;
+  line-height: normal;
+}
+
+.resultDetail {
+  display: flex;
+  flex-direction: row;
+  color: var(--lego-font-color);
+  font-size: var(--font-size-sm);
+  margin: 0;
+}
+
+.resultDetail > *:not(:last-child)::after {
```

Comment:
It looks kinda ugly, but in my opinion, it's better to utilize the features of css rather than creating a lot of classnames. This will also declutter the JSX code, and remove JSX code that is used for styling purposes only.